dask/dataframe/optimize.py

```diff
@@ -34,10 +35,12 @@ def optimize(dsk, keys, **kwargs):
     return dsk
 
 
-def optimize_read_parquet_getitem(dsk):
+def optimize_read_parquet_getitem(dsk, keys):
     # find the keys to optimize
     from .io.parquet.core import ParquetSubgraph
 
+    keys = set(keys)
```

Comment:
Thoughts on using something like `set(map(key_split, keys))` instead so that we don't have to generate the full key-sets for all of the graphs?  Or is this unlikely to be a performance issue because we only do it relatively rarely?

Reply:
For my own understanding, is this likely to be a large object in normal cases? My understanding is that the size here is generally equal to the number of collections * the number of partitions being optimized.

Earlier in `optimize`, convert this to a list with `list(core.flatten(keys))`.